### PR TITLE
update to JavaSE-11

### LINKED ISSUE #16: uploadNames and uploadLogEventData always throw an exception?
@digitaldan 
I haven't run into an issue in practice and maybe I am just missing something here.  But I think the logic is off on uploadNames and uploadLogEventData methods.
The code is [here](https://github.com/digitaldan/jomnilink/blob/master/src/main/java/com/digitaldan/jomnilinkII/Connection.java#L666-L674)

```
public Message uploadEventLogData(int number, int direction) throws IOException, OmniNotConnectedException,
			OmniInvalidResponseException, OmniUnknownMessageTypeException {
		Message msg = sendAndReceive(new UploadEventRecord(number, direction));
		if (msg.getMessageType() != Message.MESG_TYPE_EVENT_LOG_DATA
				&& msg.getMessageType() != Message.MESG_TYPE_END_OF_DATA) {
			throw new OmniInvalidResponseException(msg);
		}
		return msg;
	}
```
Doesn't that check on message type need to be a logical or rather than a logical and?  The returned method type is never going to be both types, so it would always throw an exception, right?

### PR DESCRIPTION
**Pull request by ecdye**
Signed-off-by: Ethan Dye <mrtops03@gmail.com>

### CODE REVIEW

File: src/main/java/com/digitaldan/jomnilinkII/Connection.java [hunk 1]
```diff
@@ -518,7 +518,7 @@ public Message reqObjectProperties(int objectType, int objectNum, int direction,
 													.build();
 		Message msg = sendAndReceive(reqObjectProperties);
 		if (msg.getMessageType() != Message.MESG_TYPE_OBJ_PROP
-				&& msg.getMessageType() != Message.MESG_TYPE_END_OF_DATA) {
+				|| msg.getMessageType() != Message.MESG_TYPE_END_OF_DATA) {
```
Comment (digitaldan) [inline src/main/java/com/digitaldan/jomnilinkII/Connection.java]: Can you explain what the issue is here ?  This code has been around for a very long time and i'm trying to understand what this is fixing, and why it has not shown up before.

Comment (digitaldan) [reply src/main/java/com/digitaldan/jomnilinkII/Connection.java]: nm, i see the original code   does make sense at all, i'm not sure why this never bubbled up as an issue before, strange.

**Comment by digitaldan** [comment]
so these changes break the library for me, when i try running the basic example from the README , i get the following error, running master does not have the error:

```
java  -cp ./target/jomnilink-1.3.7-jar-with-dependencies.jar  com.digitaldan.jomnilinkII.examples.Main  192.168.0.1 4369 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
[main] INFO com.digitaldan.jomnilinkII.examples.Main - SystemInformation(model=38, major=4, minor=0, revision=2, phone=-)
[main] INFO com.digitaldan.jomnilinkII.examples.Main - SystemStatus(timeDateValid=true, year=20, month=11, day=2, dayOfWeek=1, hour=6, minute=19, second=36, daylightSavings=false, sunriseHour=6, sunriseMinute=0, sunsetHour=17, sunsetMinute=11, batteryReading=170, alarms={})
[main] INFO com.digitaldan.jomnilinkII.examples.Main - SystemTroubles(troubles=[2])
[main] INFO com.digitaldan.jomnilinkII.examples.Main - SystemFormats(tempFormat=1, timeFormat=1, dateFormat=1)
[main] INFO com.digitaldan.jomnilinkII.examples.Main - SystemFeatures(features=[])
[main] INFO com.digitaldan.jomnilinkII.examples.Main - ObjectTypeCapacities(objectType=9, capacity=8)
[main] INFO com.digitaldan.jomnilinkII.examples.Main - ObjectTypeCapacities(objectType=10, capacity=8)
[main] ERROR com.digitaldan.jomnilinkII.examples.Main - Invalid Response
com.digitaldan.jomnilinkII.OmniInvalidResponseException
	at com.digitaldan.jomnilinkII.Connection.reqObjectProperties(Connection.java:522)
	at com.digitaldan.jomnilinkII.examples.Main.main(Main.java:133)
```

**Comment by ecdye** [comment]
@digitaldan I figured it out, apparently your original code was correct because of the way java boolean logic works. I have reverted those changes and it appears to be working now. As a result this PR essentially only upgrades the JavaSE version to 11.

**Comment by digitaldan** [comment]
LGTM, thanks.

**Pull request merged**